src/rmutil/args.c

```diff
@@ -0,0 +1,125 @@
+#include "args.h"
+#include "redismodule.h"
+#include <float.h>
+#include <math.h>
+#include <string.h>
+
+int AC_Advance(ArgsCursor *ac) {
+  if (ac->offset == ac->argc) {
+    return AC_ERR_NOARG;
+  } else {
+    ac->offset++;
+  }
+  return 0;
+}
+
+#define MAYBE_ADVANCE()            \
+  if (!(flags & AC_F_NOADVANCE)) { \
+    AC_Advance(ac);                \
+  }
+
+static int tryReadAsDouble(ArgsCursor *ac, long long *ll, int flags) {
+  double dTmp = 0.0;
+  if (AC_GetDouble(ac, &dTmp, flags | AC_F_NOADVANCE)) {
+    return AC_ERR_PARSE;
+  }
+  if (flags & AC_F_COALESCE) {
+    *ll = dTmp;
+    return 0;
+  }
+
+  if ((double)(long long)dTmp != dTmp) {
+    return AC_ERR_PARSE;
+  } else {
+    *ll = dTmp;
+    return 0;
+  }
+}
+
+int AC_GetLongLong(ArgsCursor *ac, long long *ll, int flags) {
+  if (ac->offset == ac->argc) {
+    return AC_ERR_NOARG;
+  }
+  if (ac->type == AC_TYPE_RSTRING) {
+    if (RedisModule_StringToLongLong(AC_CURRENT(ac), ll) == REDISMODULE_ERR &&
+        tryReadAsDouble(ac, ll, flags)) {
```

Comment:
Shouldn't there be a return AC_ERR_NOARG here? or something else?

Reply:
So why do you need an if statement if there is nothing inside the if?

Reply:
Fixed